### PR TITLE
PPS pixel topology

### PR DESCRIPTION
**Pull request by fabferro**
#### PR description:
The PPS Pixel topology (internal sensor segmentation) was hardwired in the code and prevented the coexistence in the same release of different types of pixel detectors (eg. Run2 and Run3 ones). For testing and validation puposes it's important for PPS to be able to process both Run2 and Run3 data and simulation. This PR removes the old files and allows to manage the detector topology via the EventSetup, so that the same code can be used to processed data taken or simulated with different pixel detectors. The corresponding setup is chosen by means of the Era mechanism.

No changes in the output are expected.

This PR will be followed by the implementation of the new (Run3) detector Geometry.

#### PR validation:

Private tests with real data and simulation show no difference in the output of PPS detectors.
runTheMatrix tests are successful.

No backport is expected.

### CODE REVIEW

**Comment by fabferro** [comment]
@jan-kaspar @mundim FYI

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1
No DQM changes as expected

**Comment by cvuosalo** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by chayanit** [comment]
+1

**Comment by yuanchao** [comment]
+1
* WF matrix tests and unit tests all passed
* static build warnings are not of this PR

**Comment by srimanob** [comment]
+Upgrade

File: RecoPPS/Local/plugins/CTPPSPixelRecHitProducer.cc [hunk 1]
```diff
@@ -20,22 +21,27 @@ void CTPPSPixelRecHitProducer::produce(edm::Event &iEvent, const edm::EventSetup
...
   edm::DetSetVector<CTPPSPixelRecHit> output;
 
   // run reconstruction
   if (!rpCl->empty())
-    run(*rpCl, output);
+    run(*rpCl, output, *thePixelTopology);
 
   iEvent.put(std::make_unique<edm::DetSetVector<CTPPSPixelRecHit> >(output));
+
+  //  std::cout << (*thePixelTopology) << std::endl;
```
Comment (jpata) [inline RecoPPS/Local/plugins/CTPPSPixelRecHitProducer.cc:34]: this might be removed

**Review by jpata — COMMENTED** [review]
a minor reco comment inline (remove unneeded commented out `cout`)

File: RecoPPS/Local/plugins/CTPPSPixelRecHitProducer.cc (hunk 1 above)
Comment (fabferro) [reply RecoPPS/Local/plugins/CTPPSPixelRecHitProducer.cc:34]: Sure. Removed. Waiting for more comments before pushing the code.

**Comment by ggovi** [comment]
+1

**Comment by srimanob** [comment]
please test

Don't expect any issue, just removing a comment line.

**Comment by jfernan2** [comment]
+1
There changes in L1TEMU / L1TdeStage2uGMT / data_vs_emulator_comparison
for wf 11634.911 and 312.0  but they are unrelated to this PR

**Comment by cvuosalo** [comment]
+1

**Comment by jpata** [comment]
+reconstruction
- technical, loads the PPS topology from ES instead of hardcoding (reco changes are minimal)
- no observed reco differences, as expected

**Comment by yuanchao** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

Please make sure to cross-check when pre4 relvals will be available (validation vs pre3).

**Comment by fabferro** [comment]
@silviodonato : hoping this PR can be merged soon, so that I can proceed with the implementation of the new Run3 geometry. Thanks!

**Comment by silviodonato** [comment]
@cms-sw/pdmv-l2 @cms-sw/db-l2 do you have any comments?

**Comment by chayanit** [comment]
+1 
I had already signed

**Comment by silviodonato** [comment]
merge
@ggovi please have a look, especially at  CondFormats/PPSObjects

**Pull request merged**